packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,48 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number | string) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
+      onChange(val);
+    },
+    [onChange]
+  );
+
+  const appendField = useCallback(
+    (fieldId: string) => {
+      onChange(fieldId);
+    },
+    [onChange]
+  );
+
+  return (
+    <FieldPickerInput
+      onInputChange={handleChangeEvent}
+      handleFieldSelection={appendField}
```

Comment:
nit: can `onChange` be passed directly here instead of creating a new `appendField` function?

Reply:
Hmmm yes, but it looks messier to me? 
<img width="317" alt="Screenshot 2023-08-23 at 9 41 15 AM" src="https://github.com/yext/studio/assets/106991746/8e7685f1-2c1d-4bb3-ac90-dcc8bc394822">

Reply:
oh I meant something like this:
```tsx
<FieldPickerInput
  // ... other props
  handleFieldSelection={onChange}
/>
```

Reply:
Oh like removing the useCallback? When I do I get an autofix warning `JSX attribute values should not contain functions created in the same scope`

Reply:
you get that error if you use the code snippet above? I'm pretty sure that's how we pass it in ArrayPropEditor

Reply:
If I use this `handleFieldSelection={fieldId => onChange(fieldId)}`

Reply:
ah that makes sense since that's creating a new function in line. you should be able to directly pass the function (`handleFieldSelection={onChange}`) since `onChange` is passed in as a prop to the component, rather than created in the body of the component

Reply:
Ohhh I see! Thank you!